apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,193 @@
+          credit: req.body.credit,
+          end_date: new Date(req.body.end_date),
+          group_name: req.body.group_name || null,
+          note: req.body.note || null,
+          start_date: new Date(req.body.start_date),
+          group_id: group_id || null,
+          user_id: user_id || null,
+        }
+      });
+    
+      res.redirect(req.originalUrl);
+    } else if (
+      req.body.__action === 'delete_override' &&
+      res.locals.authz_data.has_course_instance_permission_edit
+    ) {
+      const deletedRow = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+        assessment_id: res.locals.assessment.id,
+        assessment_access_policies_id: req.body.policy_id,
+      });
+      await sqldb.queryAsync(sql.delete_assessment_access_policy, {
+        assessment_id: res.locals.assessment.id,
+        unsafe_assessment_access_policies_id: req.body.policy_id,
+      });
+      await insertAuditLog({
+        authn_user_id: res.locals.authn_user.user_id,
+        table_name: 'assessment_access_policies',
+        action: 'delete',
+        institution_id: res.locals.institution.id,
+        course_id: res.locals.assessment.id,
+        course_instance_id: res.locals.course_instance.id,
+        old_state: JSON.stringify(deletedRow.rows),
+      });
+      
+      res.redirect(req.originalUrl);
+    } else if (
+      req.body.__action === 'edit_override' &&
+      res.locals.authz_data.has_course_instance_permission_edit
+    ) {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+      await sqldb.queryAsync(sql.update_assessment_access_policy, {
+        assessment_id: res.locals.assessment.id,
+        credit: req.body.credit,
+        end_date: new Date(req.body.end_date),
+        group_name: req.body.group_name || null,
+        note: req.body.note || null,
+        start_date: new Date(req.body.start_date),
+        group_id: group_id || null,
+        user_id: user_id || null,
+        assessment_access_policies_id: req.body.policy_id,
+      });
+      
+      await insertAuditLog({
+        authn_user_id: res.locals.authn_user.user_id,
+        table_name: 'assessment_access_policies',
+        action: 'edit',
+        institution_id: res.locals.institution.id,
+        course_id: res.locals.assessment.id,
+        course_instance_id: res.locals.course_instance.id,
+        new_state: JSON.stringify({
+          assessment_id: res.locals.assessment.id,
+          credit: req.body.credit,
+          end_date: new Date(req.body.end_date),
+          group_name: req.body.group_name || null,
+          note: req.body.note || null,
+          start_date: new Date(req.body.start_date),
+          group_id: group_id || null,
+          user_id: user_id || null,
+        }),
```

Comment:
As above, you should add `RETURNING *` to the `update_assessment_access_policy` query and use the resulting row here.

Don't forget to include a value for `old_state` here, as well as `row_id` with the updated row's id.

Reply:
done in [a046c89](https://github.com/PrairieLearn/PrairieLearn/pull/9103/commits/a046c897ce189522e2d80ea862b69070929e2f65)

Reply:
done in [a046c89](https://github.com/PrairieLearn/PrairieLearn/pull/9103/commits/a046c897ce189522e2d80ea862b69070929e2f65)